source/jormungandr/jormungandr/scenarios/journey_filter.py

```diff
@@ -200,84 +214,82 @@ def exceed_min_duration(current_section, journey, request,  min_duration):
            and current_section.duration < min_duration
 
 
-def _filter_too_short_heavy_journeys(journeys, request):
+def filter_too_short_heavy_journeys(journey, request):
     """
     we filter the journeys with use an "heavy" fallback mode if it's use only for a few minutes
     Heavy fallback mode are Bike and Car, bss is not considered as one.
     Typically you don't take your car for only 2 minutes
     """
     logger = logging.getLogger(__name__)
-    for journey in journeys:
-        if to_be_deleted(journey):
+
+    on_bss = False
+    is_debug = request.get('debug', False)
+    for s in journey.sections:
+        if s.type == response_pb2.BSS_RENT:
+            on_bss = True
+        elif s.type == response_pb2.BSS_PUT_BACK:
+            on_bss = False
+        elif s.type != response_pb2.STREET_NETWORK:
             continue
-        on_bss = False
-        for s in journey.sections:
-            if s.type == response_pb2.BSS_RENT:
-                on_bss = True
-            if s.type == response_pb2.BSS_PUT_BACK:
-                on_bss = False
-            if s.type != response_pb2.STREET_NETWORK:
-                continue
-
-            if s.street_network.mode == response_pb2.Car \
-                    and exceed_min_duration(s, journey, request, request['_min_car']):
-                logger.debug("the journey {} has not enough car, we delete it".format(journey.internal_id))
-                mark_as_dead(journey, "not_enough_car")
-                break
-            if not on_bss \
-                    and s.street_network.mode == response_pb2.Bike \
-                    and exceed_min_duration(s, journey, request, request['_min_bike']):
-                logger.debug("the journey {} has not enough bike, we delete it".format(journey.internal_id))
-                mark_as_dead(journey, "not_enough_bike")
-                break
-
-
-def _filter_too_long_waiting(journeys, request):
+
+        if s.street_network.mode == response_pb2.Car \
+                and exceed_min_duration(s, journey, request, request['_min_car']):
+            logger.debug("the journey {} has not enough car, we delete it".format(journey.internal_id))
+            mark_as_dead(journey, is_debug, "not_enough_car")
+            return False or is_debug
+        if not on_bss \
+                and s.street_network.mode == response_pb2.Bike \
+                and exceed_min_duration(s, journey, request, request['_min_bike']):
+            logger.debug("the journey {} has not enough bike, we delete it".format(journey.internal_id))
+            mark_as_dead(journey, is_debug, "not_enough_bike")
+            return False or is_debug
+    return True
+
+
+def filter_too_long_waiting(journey, is_debug):
     """
     filter journeys with a too long section of type waiting
     """
     logger = logging.getLogger(__name__)
-    for j in journeys:
-        if to_be_deleted(j):
+
+    for s in journey.sections:
```

Comment:
Same trick here, if the duration of the journey is less than max_duration there is no need to check each sections. 
Also I think we have the number of transfers of a journey, if there is no transfers it won't have any waiting sections

Reply:
OK for this one